presto-parquet/src/main/java/com/facebook/presto/parquet/predicate/PredicateUtils.java

```diff
@@ -105,11 +106,13 @@ private static Map<ColumnDescriptor, Statistics<?>> getStatistics(BlockMetaData
     {
         ImmutableMap.Builder<ColumnDescriptor, Statistics<?>> statistics = ImmutableMap.builder();
         for (ColumnChunkMetaData columnMetaData : blockMetadata.getColumns()) {
-            Statistics<?> columnStatistics = columnMetaData.getStatistics();
-            if (columnStatistics != null) {
-                RichColumnDescriptor descriptor = descriptorsByPath.get(Arrays.asList(columnMetaData.getPath().toArray()));
-                if (descriptor != null) {
-                    statistics.put(descriptor, columnStatistics);
+            if (!HiddenColumnChunkMetaData.isHiddenColumn(columnMetaData)) {
+                Statistics<?> columnStatistics = columnMetaData.getStatistics();
+                if (columnStatistics != null) {
```

Comment:
NIT: Merge those two ifs, too many nested statements hurt code readability

Reply:
Between these two diffs, we can save the .toArray() call if the first 'if' is not true. But fine, not a big deal anyway. I just merged it.